04-backend-api.md

```diff
@@ -0,0 +1,191 @@
+
+Start by changing from the *Visua*l to *Template* editor mode by clicking on **Template** in the upper left. In the template editor find the *Api* resource (it will be named *Api* and have a *Type* attribute with the value `AWS::Serverless::Api`). You can use *Ctrl+f* to search the file for this. Add the following authentication configuration under the *Properties* key.
+
+```YAML
+Auth:
+  Authorizers:
+  	WildRydes:
+    	UserPoolArn: !GetAtt UserPool.Arn
+```
+
+Next locate the *POST /ride* route under *DefinitionBody -> paths -> /ride -> post* path in the YAML and add the following *security* property:
+```YAML
+security:
+    - WildRydes: []
+```
+The complete *Api* resource definition look like it doies below. *(NOTE: that the order of the properties doesn't matter)*
+```YAML
+  Api:
+    Type: AWS::Serverless::Api
+    Properties:
+      Auth:
+        Authorizers:
+          WildRydes:
+            UserPoolArn: !GetAtt UserPool.Arn
+      Name: !Sub
+        - ${ResourceName} From Stack ${StackTagName} Environment ${EnvironmentTagName}
+        - ResourceName: Api
+      StageName: !Ref EnvironmentAPIGatewayStageName
+      DefinitionBody:
+        swagger: '2.0'
+        info: {}
+        paths:
+          /ride:
+            post:
+              security:
+                - WildRydes: []
+              x-amazon-apigateway-integration:
+                httpMethod: POST
+                type: aws_proxy
+                uri: !Sub arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${RequestUnicorn.Arn}/invocations
+              responses: {}
+      EndpointConfiguration: REGIONAL
+      Cors:
+        AllowOrigin: '''*'''
+        AllowHeaders: '''Authorization,Content-Type'''
+
+```
+
+Finally, commit your changes by clicking the **Commit** button and in the popup window clicking **Commit and Push**.
+
+![Commit And Push](./images/04-commit-and-push.png)
+
+
+
+### 7. Update *RequestUnicorn* Function.
+
+Update the *RequestUnicorn* Function code so it is functional. The code accepts requests and performs the following actions:
+
+* Validates the User Pool Client authentication token (provided via the *Authorization* header from API Gateway) and retrieves the username from it
+* Retrieves the Unicorn Stables™ api key from AWS Secrets Manager and caches the value for subsequent requests
+* Makes a request to the Unicorn Stables™ endpoint to rent a unicorn
+* Records the ride to the *Rides* DynamoDB table
+* Returns with the response to the frontend website request
+
+Start by updating your local *stackery-wild-rydes* GitHub clone (created in module 1) so it has all the changes you've made over the past few workshop modules.
+
+```
+$ cd stackery-wild-rydes     # if not already in the directory.
+$ git fetch origin
+$ git rebase origin/master
```

Comment:
The rebase is needed because of the committed changes in modules 2 and 3. Changing instructions though.

resolved in 488e1704ae1f90e912079722ee49b16cd655ba1a